jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ManifestListGenerator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.image.json;
+
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.blob.Blobs;
+import com.google.cloud.tools.jib.configuration.BuildContext;
+import com.google.cloud.tools.jib.image.Image;
+import com.google.cloud.tools.jib.image.json.V22ManifestListTemplate.ManifestDescriptorTemplate;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.util.List;
+
+/** Translates a list of {@link Image} into a manifest list. */
+public class ManifestListGenerator {
+
+  private final BuildContext buildContext;
+  private final List<Image> builtImages;
+
+  public ManifestListGenerator(BuildContext buildContext, List<Image> builtImages) {
```

Comment:
From the API design perspective, it shouldn't know anything about `BuildContext`. It's an isolated class and shouldn't be coupled.

```suggestion
  public ManifestListGenerator(... imageFormat, List<Image> images) {
```